api/envoy/extensions/filters/http/oauth2/v3alpha/oauth.proto

```diff
@@ -74,6 +74,9 @@ message OAuth2Config {
 
   // Any request that matches any of the provided matchers will be passed through without OAuth validation.
   repeated config.route.v3.HeaderMatcher pass_through_matcher = 8;
+
+  // Space separated list of scopes to be claimed in the authorization request
```

Comment:
Why space separate rather than make this a `repeated string`?

Reply:
@htuch basically my lack of knowledge of how to handle this in proto. And also because I've been seen other OAuth solutions that use space separate as a solution, so I had that in mind.
If you want to request the change that's fine, I'll work on that. Any guidance would be much appreciated

Reply:
I guess this betrays my lack of knowledge on the OAuth space; generally in proto we prefer structure, but if there is some standard format then this is OK. CC @snowp

Reply:
Can you please add the default value ("user") in the description?
Can this be an empty string?

Reply:
@adisuissa I've just added the default value in the comment, thanks for your suggestion
if the property is not provided, it will be compared to an empty string in the FilterConfig getter property. Then it will default to 'user' in this case since 'scope' is a mandatory parameter for OAuth

Reply:
I would also prefer this to be a repeated field for a bit more obvious API.

 It seems like the space delimited string (URL encoded) is passed as a query param, which is probably where this format comes from. 

Maybe @williamsfu99 has thoughts here?

Reply:
+1 to repeated field. Space-delimitation is from [the OAuth RFC](https://tools.ietf.org/html/rfc6749#section-3.3) for the actual request made to the authorization server, but that can just be a `absl::StrJoin` join on the scopes

Reply:
I'm currently working to convert this to repeated string. Does anybody know if default value would be possible in proto? like this: 
`optional repeated string auth_scopes = 9 [default = "user"];`
this would be a nice way to get rid of the default logic in the filters code.
tried this syntax but `proto_format.sh fix` is failing (not sure why)

Reply:
To the best of my knowledge in proto3 you don't need to set the field as optional (it will be optional).
Regarding the default value, I don't think there are custom default values, and this should be implemented in the code (i.e., if the list is empty, add "user" to it).

Reply:
I'm working to implement `repeated string auth_scopes = 9;`, but having issues with the generated cpp code. Looks like the accessors described in https://developers.google.com/protocol-buffers/docs/reference/cpp-generated#repeatedstring are not available. 
Posted a question in Slack's #envoy-users channel, if anyone have time to check it I'd appreciate

Reply:
You probably need to re-run `./tools/proto_format/proto_format.sh fix` as the actual protos are read from the "generated_api_shadow" directory (see [here](https://github.com/envoyproxy/envoy/blob/master/api/STYLE.md#adding-an-extension-configuration-to-the-api)).

Reply:
Instead of special casing 'scope', would it make sense to make this:
 map<string, string> additional_authorize_parameters = 9 ?

This would allow passing other parameters (such as audience) using the same mechanism

Reply:
Thanks @adisuissa, this fixed the issue and now the code is compilling.
But now I'm facing an error to run automated tests, even in the upstream master branch or in my feature branch after rebasing it (this was not happening before).

This error is thrown in both master as well, without any changes added: `ERROR: /source/test/mocks/server/BUILD:234:14: C++ compilation of rule '//test/mocks/server:transport_socket_factory_context_mocks' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer '-std=c++0x' -MD -MF ... (remaining 580 argument(s) skipped)`.

I'm running the tests for oauth filter only as follows: `./ci/run_envoy_docker.sh 'bazel test test/extensions/filters/http/oauth2/...'`

Reply:
I've just updated to the latest master and tried `./ci/run_envoy_docker.sh 'bazel test test/extensions/filters/http/oauth2/...'` (without your changes) and it works.
I don't see the entire error from your comment. Would you be able to copy-paste it to a slack chat (my user is "Adi Peleg" in the Envoy slack)?